extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -190,6 +191,48 @@ function () {
 		return $botIds;
 	}
 
+
+	/**
+	 * @return array list of blacklisted ids for Top Contributors
+	 */
+	private function getBlacklistedIds(){
+		$blacklistedIds = WikiaDataAccess::cache(
+			self::getMemcKey( self::ALL_BLACKLISTED_IDS_MCACHE_KEY ),
+			WikiaResponse::CACHE_LONG,
+			function () {
+				global $wgExternalSharedDB;
+				$globalDb = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+				$globalIds = ( new WikiaSQL() )
+					->SELECT( 'ug_user' )
+					->FROM ( 'user_groups' )
+					->WHERE( 'ug_group' )->IN( [ 'bot', 'bot-global', 'staff', 'util', 'helper', 'vstf' ] )
+					->GROUP_BY( 'ug_user' )
+					->runLoop( $globalDb, function ( &$globalIds, $row ) {
+						$globalIds[] = $row->ug_user;
+					} );
+
+				$localDb = wfGetDB( DB_SLAVE );
+
+				$localUsers = ( new WikiaSQL() )
+					->SELECT( 'ug_user' )
+					->FROM ( 'user_groups' )
+					->WHERE( 'ug_group' )->NOT_IN( [ 'bot', 'bot-global' ] )
```

Comment:
Isn't this just [the exact query from `getBotIds`](https://github.com/Wikia/app/blob/6da13d7280cb58024838318061bc83933cd2d6f5/extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php#L174-L185), so would be redundant to calling `getBotIds` below, right?